### ISSUE TITLE
FastSpeech2 loss=NaN in early iterations

### ISSUE THREAD

**Issue by tekinek** [TensorSpeech/TensorFlowTTS#531]
Thanks for your great work. It has been a while since my last experiment with TensorFlowTTS. The FastSpeech2 was fine for my private dataset of local language. But with the recent codebase and baker dataset, loss values in FS2 turn NaN at early stage of training and show no sign for going back to normal:

<img width="1131" alt="nan-error" src="https://user-images.githubusercontent.com/68597606/113039694-87a39c80-91ca-11eb-87bb-b945a8225718.png">

FYI
- Preprocessing was done under default baker setting.
- Tacotron2 was trained (with no problem) and used to extract durations.
- mixed_precision = 0
- _fastspeech2.baker.v2.yaml_ and other configs didn't solve the problem.
- Different learning rates and weight decay didn't solve the problem.
- Tensorflow = 2.2
- (No problem with FastSpeech 1 )

Any comment would be appriciated.

**Comment by dathudeptrai** [comment]
@tekinek can you try to using newer tensorflow version first ?

**Comment by tekinek** [comment]
@dathudeptrai thanks, let me try tf2.3.

**Comment by dathudeptrai** [comment]
> @dathudeptrai thanks, let me try tf2.3.

Next step is check forward pass to see if it yield nan output or not. To check the output you can refer this simple code:

```
for data in tqdm(train_dataset):
    outs = fastspeech2(**data)
    [CODE TO CHECK NAN HERE]
    [you can use .numpy() then use numpy to check or `tf.debugging.check_numerics`]
```

**Comment by tekinek** [comment]
@dathudeptrai sorry, but Tensorflow 2.3 couldn't fix the problem:

![nan-error2](https://user-images.githubusercontent.com/68597606/113092085-ed6e4380-921f-11eb-8016-40f40287aff2.jpg)

**Comment by tekinek** [comment]
@dathudeptrai  a NaN value in a f0 file seems to be the source of problem. The utters number 9678 and 8920 contain NaN values (I don't know why). Skipping them solves the problem (so far at 5000 iters).

For anyone with similar problem, please use the following code snippets to check your data against NaN value.


```
import os
from os import path
import numpy as np

dump_location = "/path/to/your/dump/directory/train/"
for directory in ["raw-f0", "raw-energies"]:
    for file_name in os.listdir(path.join(dump_location, directory)):
        if True in np.isnan(np.load(path.join(dump_location, directory, file_name))):
            print('{} has NaN value'.format(file_name))
```

Thanks.
